tests/ui/statics/nested_thread_local.rs

```diff
@@ -0,0 +1,14 @@
+// Check that we forbid nested statics in `thread_local` statics.
+
+#![feature(const_refs_to_cell)]
+#![feature(thread_local)]
+
+#[thread_local]
```

Comment:
Does this not also affect `thread_local!()` macro invokations? It should, since those just desugar to `#[thread_local]` internally?

Reply:
yes, but those always desugar to a `Key<StaticsType>`, which is always initialized with `None` internally, and then set to `Some` by computing a value at runtime at thread start.

Reply:
I'm looking at the implementation of `thread_local_inner` in `library/std/src/sys/thread_local/fast_local.rs`, which initializes:

```

            #[thread_local]
            // We use `UnsafeCell` here instead of `static mut` to ensure any generated TLS shims
            // have a nonnull attribute on their return value.
            static VAL: $crate::cell::UnsafeCell<$t> = $crate::cell::UnsafeCell::new(INIT_EXPR);
```

Reply:
:facepalm: of course!